src/providers/strava.js

```diff
@@ -4,17 +4,54 @@ export default function Strava(options) {
     id: "strava",
     name: "Strava",
     type: "oauth",
-    authorization: "https://www.strava.com/api/v3/oauth/authorize?scope=read",
-    token: "https://www.strava.com/api/v3/oauth/token",
+    authorization: {
+      url: "https://www.strava.com/api/v3/oauth/authorize",
+      params: {
+        scope: "read",
+        approval_prompt: "auto",
+        response_type: "code",
+        redirect_uri: "http://localhost:3000/api/auth/callback/strava"
+      }
+    },
+    token: { 
+      url: "https://www.strava.com/api/v3/oauth/token",
+      async request(context) {
+        const response = await fetch(context.provider.token.url, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            ...context.params,
+            client_id: process.env.STRAVA_CLIENT_ID,
+            client_secret: process.env.STRAVA_CLIENT_SECRET,
+            grant_type: "authorization_code",
+          }),
+        });
+
+        const { refresh_token, access_token, token_type, expires_at } = await response.json();
+
+        return {
+          tokens: {
+            refresh_token,
+            access_token,
+            token_type, 
+            expires_at,
+          },
+        };
```

Comment:
This part is most certainly unnecessary. If the configuration for Strava does not match the defaults, maybe the `client` provider config option might be of use.

https://next-auth.js.org/configuration/providers/oauth#client-option

Reply:
@krautwigundrueben you can just add:

```js
client: {
  token_endpoint_auth_method: "client_secret_post",
}
```